theano/tensor/signal/pool.py

```diff
@@ -540,9 +540,9 @@ def make_node(self, x, ws, stride=None, pad=None):
                 raise NotImplementedError(
                     'padding works only with ignore_border=True')
             if isinstance(ws, (tuple, list)):
-                if any(pad[i] >= ws[i] for i in range(nd)):
-                    raise NotImplementedError(
-                        'padding must be smaller than strides')
+                if not (isinstance(pad[i], theano.Variable) or isinstance(ws[i], theano.Variable) for i in range(nd)):
```

Comment:
The test isn't good. It should be some think like:
for i in range(nd):
   if not isinstance(pad[i], ...) and not isinstance(ws[i], ...):
      if pad[i] >= ws[i]:
          raise